### ISSUE TITLE
gracefully handle invalid data type form JSON

### ISSUE THREAD

**Issue by jsvd** [logstash-plugins/logstash-codec-json_lines#27]
While the codec properly recovers from invalid JSON, if fed an array it crashes the pipeline:

```
logstash-2.3.2 % echo -e "[1]\n" | bin/logstash -e 'input { stdin { codec => json_lines } } output { stdout { codec => json } }' 
Pipeline main started
A plugin had an unrecoverable error. Will restart this plugin.
  Plugin: <LogStash::Inputs::Stdin codec=><LogStash::Codecs::JSONLines charset=>"UTF-8", delimiter=>"\n">>
  Error: can't convert String into Integer {:level=>:error}
Pipeline main has been shutdown
```

```
logstash-2.3.2 % echo -e "1]\n" | bin/logstash -e 'input { stdin { codec => json_lines } } output { stdout { codec => json } }' 
Settings: Default pipeline workers: 4
Pipeline main started
{"message":"1]","tags":["_jsonparsefailure"],"@version":"1","@timestamp":"2016-06-03T07:45:53.982Z","host":"Joaos-MBP.lan"}
Pipeline main has been shutdown
```

**Comment by colinsurprenant** [comment]
just checked and both the examples above dramatically crashes logstash on master today. So this is still a valid bug.

**Comment by kares** [comment]
seem this has been resolved since, in 7.13.0 both examples properly tag the event with `"_jsonparsefailure"`
the plugin no longer crashes ...

```
WARNING, using JAVA_HOME while Logstash distribution comes with a bundled JDK
Sending Logstash logs to /home/kares/workspace/work/elastic/logstash-7.13.0/logs which is now configured via log4j2.properties
[2021-06-29T10:01:00,242][INFO ][logstash.runner          ] Log4j configuration path used is: /home/kares/workspace/work/elastic/logstash-7.13.0/config/log4j2.properties
[2021-06-29T10:01:00,439][WARN ][logstash.runner          ] Logging configuration uses appender or filter with scripting language JavaScript, which will be removed in a future major release of Logstash.
[2021-06-29T10:01:00,450][INFO ][logstash.runner          ] Starting Logstash {"logstash.version"=>"7.13.0", "jruby.version"=>"jruby 9.2.16.0 (2.5.7) 2021-03-03 f82228dc32 Java HotSpot(TM) 64-Bit Server VM 25.171-b11 on 1.8.0_171-b11 +indy +jit [linux-x86_64]"}
[2021-06-29T10:01:01,117][WARN ][logstash.config.source.multilocal] Ignoring the 'pipelines.yml' file because modules or command line options are specified
[2021-06-29T10:01:02,906][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2021-06-29T10:01:04,078][INFO ][org.reflections.Reflections] Reflections took 79 ms to scan 1 urls, producing 24 keys and 48 values 
[2021-06-29T10:01:05,952][INFO ][logstash.javapipeline    ][main] Starting pipeline {:pipeline_id=>"main", "pipeline.workers"=>16, "pipeline.batch.size"=>10, "pipeline.batch.delay"=>50, "pipeline.max_inflight"=>160, "pipeline.sources"=>["config string"], :thread=>"#<Thread:0x4b933273 run>"}
[2021-06-29T10:01:07,425][INFO ][logstash.javapipeline    ][main] Pipeline Java execution initialization time {"seconds"=>1.45}
[2021-06-29T10:01:07,516][INFO ][logstash.javapipeline    ][main] Pipeline started {"pipeline.id"=>"main"}
[2021-06-29T10:01:07,652][INFO ][logstash.agent           ] Pipelines running {:count=>1, :running_pipelines=>[:main], :non_running_pipelines=>[]}
[2021-06-29T10:01:07,715][WARN ][logstash.codecs.jsonlines][main][a948728571f9292e05a9e14595d114c50939c8a49a1db59b0430783ec8e4cd20] JSON parse error, original data now in message field {:error=>#<LogStash::Json::ParserError: java.lang.Integer cannot be cast to java.util.Map>, :data=>"[1]"}
{"@version":"1","tags":["_jsonparsefailure"],"@timestamp":"2021-06-29T08:01:07.739Z","message":"[1]","host":"precision"}[2021-06-29T10:01:07,986][INFO ][logstash.javapipeline    ][main] Pipeline terminated {"pipeline.id"=>"main"}
```
```
WARNING, using JAVA_HOME while Logstash distribution comes with a bundled JDK
Sending Logstash logs to /home/kares/workspace/work/elastic/logstash-7.13.0/logs which is now configured via log4j2.properties
[2021-06-29T10:06:01,318][INFO ][logstash.runner          ] Log4j configuration path used is: /home/kares/workspace/work/elastic/logstash-7.13.0/config/log4j2.properties
[2021-06-29T10:06:01,567][WARN ][logstash.runner          ] Logging configuration uses appender or filter with scripting language JavaScript, which will be removed in a future major release of Logstash.
[2021-06-29T10:06:01,579][INFO ][logstash.runner          ] Starting Logstash {"logstash.version"=>"7.13.0", "jruby.version"=>"jruby 9.2.16.0 (2.5.7) 2021-03-03 f82228dc32 Java HotSpot(TM) 64-Bit Server VM 25.171-b11 on 1.8.0_171-b11 +indy +jit [linux-x86_64]"}
[2021-06-29T10:06:02,204][WARN ][logstash.config.source.multilocal] Ignoring the 'pipelines.yml' file because modules or command line options are specified
[2021-06-29T10:06:03,860][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2021-06-29T10:06:04,733][INFO ][org.reflections.Reflections] Reflections took 68 ms to scan 1 urls, producing 24 keys and 48 values 
[2021-06-29T10:06:06,459][INFO ][logstash.javapipeline    ][main] Starting pipeline {:pipeline_id=>"main", "pipeline.workers"=>16, "pipeline.batch.size"=>10, "pipeline.batch.delay"=>50, "pipeline.max_inflight"=>160, "pipeline.sources"=>["config string"], :thread=>"#<Thread:0x4814bccd run>"}
[2021-06-29T10:06:07,769][INFO ][logstash.javapipeline    ][main] Pipeline Java execution initialization time {"seconds"=>1.29}
[2021-06-29T10:06:07,863][INFO ][logstash.javapipeline    ][main] Pipeline started {"pipeline.id"=>"main"}
[2021-06-29T10:06:08,031][INFO ][logstash.agent           ] Pipelines running {:count=>1, :running_pipelines=>[:main], :non_running_pipelines=>[]}
[2021-06-29T10:06:08,042][WARN ][logstash.codecs.jsonlines][main][a948728571f9292e05a9e14595d114c50939c8a49a1db59b0430783ec8e4cd20] JSON parse error, original data now in message field {:error=>#<LogStash::Json::ParserError: Unexpected character (']' (code 93)): Expected space separating root-level values
 at [Source: (String)"1]"; line: 1, column: 3]>, :data=>"1]"}
{"@version":"1","@timestamp":"2021-06-29T08:06:08.110Z","host":"precision","message":"1]","tags":["_jsonparsefailure"]}{"@version":"1","@timestamp":"2021-06-29T08:06:08.227Z","host":"precision"}[2021-06-29T10:06:08,460][INFO ][logstash.javapipeline    ][main] Pipeline terminated {"pipeline.id"=>"main"}

```
